internal/backend/remote-state/s3/backend.go

```diff
@@ -230,6 +230,13 @@ func (b *Backend) ConfigSchema() *configschema.Block {
 					Attributes: assumeRoleFullSchema().SchemaAttributes(),
 				},
 			},
+
+			"use_legacy_workflow": {
+				Type:        cty.Bool,
+				Optional:    true,
+				Description: "Use the legacy authentication workflow, preferring environment variables over backend configuration.",
+				Deprecated:  true,
```

Comment:
Since we won't be removing it until v1.8, we can probably hold off on deprecating it until v1.7